File: 0328-odd-even-linked-list/0328-odd-even-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
        if(head == NULL || head->next == NULL) return head;
        int i =3 ;
        ListNode* h1 = head;  ListNode* h2 = head->next;
        ListNode* temp1 = h1;   ListNode* temp2 = h2; 
        ListNode* ptr = head->next->next;
        while(ptr != NULL){
            if(i%2 == 1){
                temp1->next = ptr;
                 temp1 = ptr;  
            }else{
                temp2->next = ptr;
                 temp2 = ptr; 
            }
            ptr = ptr->next; i++;
        }
        temp1->next = h2;
        temp2->next = nullptr;
        return h1;

    }
  
};

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        int idx = 1;
         ListNode ptr = head;
         ListNode evenhead = null;
         ListNode oddhead = null;
         ListNode ptr1= null , ptr2 = null;
        while(ptr!= null){
          if(idx%2 == 0){
            if(evenhead == null){
                evenhead = ptr;
                ptr1 = evenhead;

            }else{
                ptr1.next = ptr;
                ptr1 = ptr1.next;
            }
          }
          else{
             if(oddhead == null){
                oddhead = ptr;
                ptr2 = oddhead;

            }else{
                ptr2.next = ptr;
                ptr2 = ptr2.next;
            }
          }
          idx++;
          ptr = ptr.next;
        }
         if (ptr1 != null) ptr1.next = null;

         if (ptr2 != null) ptr2.next = evenhead;
        return oddhead;
    }
}